grammar.js

```diff
@@ -871,7 +864,7 @@ module.exports = grammar({
     member_access_expression: $ => prec(PREC.MEMBER, seq(
       field('object', $._dereferencable_expression),
       '->',
-      field('name', $._member_name)
+      $._member_name
```

Comment:
In cases like this, we were associating the field `name` with the entire `_member_name`. But a `_member_name` includes the curly braces. I've moved the field annotation inside of the `_member_name` rule, so that it always just contains one node.